### PR TITLE
Add airflow GitHub script

### PR DESCRIPTION
**Pull request by jhtimmins**
This adds a Github counterpart to the airflow-jira script that generates comparisons between releases as well as changelogs.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: dev/airflow-github [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        commit="COMMIT"))
+
+    for issue in milestone_issues:
+        commit = None
+        for event in issue.get_events():
+            # Checks if event is a commit getting merged into master.
+            if event.event == 'merged':
+                commit = event.commit_id
+        if not commit:
+            continue
```
Comment (kaxil) [inline dev/airflow-github]: Because of this the PRs/Issues that are not merged yet won't show in the list

File: dev/airflow-github [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        description="DESCRIPTION",
+        merged="MERGED",
+        commit="COMMIT"))
+
+    for issue in milestone_issues:
+        commit = None
+        for event in issue.get_events():
+            # Checks if event is a commit getting merged into master.
+            if event.event == 'merged':
+                commit = event.commit_id
```
Comment (kaxil) [inline dev/airflow-github]: This loop is time-consuming, takes a good amount of time on my machine to load all commits.

Comment (kaxil) [reply dev/airflow-github]: I will push some improvements

Comment (kaxil) [reply dev/airflow-github]: Fixed in https://github.com/apache/airflow/pull/9270/commits/f93f0ee9832614921436e669141fc779ad5a18e6

File: dev/airflow-github [hunk 3]
```diff
@@ -93,6 +93,32 @@ def get_issue_type(issue):
...
 
 
+def get_commit_in_master_associated_with_pr(repo: git.Repo, issue: Issue) -> Optional[str]:
+    """For a PR Get find the associated merged commit & return is SHA"""
+    if issue.pull_request:
+        commit = repo.git.log(f"--grep=#{issue.number}", "origin/master", "--format=%H")
+        if commit:
+            return commit
+        else:
+            pr: PullRequest = issue.as_pull_request()
```
Comment (jhtimmins) [inline dev/airflow-github:93]: this is great. really like this optimization.

**Review by kaxil — APPROVED** [review]

**Pull request merged**